src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Eventing/ExceptionsEventSource.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions.Eventing
+{
+    [EventSource(Name = SourceName)]
+    internal sealed class ExceptionsEventSource : EventSource
+    {
+        public const int ExceptionIdEventId = 1;
```

Comment:
To my understanding, if the OnMetadata* events are not observed, we will have no way of seeing them again correct? I understand we probably want to start listening from the beginning of the app, but it would be nice to be able to enable/disable this and invalidate the cache as necessary so that the client can receive it again.

Reply:
I was thinking that we might want that, but I'm not sure how we'll want to do that just yet (e.g. command on the UDS channel to invalidate the cache, invalidate when the event source is disabled, etc). Invalidating when the event source is disable has a chicken-and-egg problem since the cache is the one that calls the event source already. I'd like to think about this more and save it for another PR.